src/istio/control/http/attributes_builder.cc

```diff
@@ -57,11 +58,46 @@ void AttributesBuilder::ExtractRequestHeaderAttributes(CheckData *check_data) {
   }
 }
 
-void AttributesBuilder::ExtractRequestAuthAttributes(CheckData *check_data) {
+void AttributesBuilder::ExtractAuthAttributes(CheckData *check_data) {
+  istio::authn::Result authn_result;
+  if (check_data->GetAuthenticationResult(&authn_result)) {
+    utils::AttributesBuilder builder(&request_->attributes);
+    if (!authn_result.principal().empty()) {
+      builder.AddString(AttributeName::kRequestAuthPrincipal,
+                        authn_result.principal());
+    }
+    if (!authn_result.peer_user().empty()) {
+      builder.AddString(AttributeName::kSourceUser, authn_result.peer_user());
+    }
+    if (authn_result.has_origin()) {
+      const auto &origin = authn_result.origin();
+      if (!origin.user().empty()) {
+        builder.AddString(AttributeName::kRequestAuthUser, origin.user());
+      }
+      if (!origin.audiences().empty()) {
+        // TODO(diemtvu): what to do with multiple aud ?. For now, just send the
+        // first one.
+        builder.AddString(AttributeName::kRequestAuthAudiences,
```

Comment:
we can join them with ","

Reply:
Possibly. Though I want to review what is the right approach here. In google, the attribute_context mentioned audiences as repeated string, but looks like mixer take a single string at the moment, I don't know should we change that behavior.